File: Node/Prisma_ORM/src/api/employeeTerritoryAPI.js
const express = require('express');
const EmployeeTerritoryService = require('../bll/services/employeeTerritoryService');
const router = express.Router();

router.get('/', async (req, res) => {
  try {
    const entities = await EmployeeTerritoryService.getAll();
    res.status(200).json(entities);
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.get('/:id', async (req, res) => {
  const { id } = req.params;
  try {
    const entity = await EmployeeTerritoryService.getById(Number(id));
    res.status(200).json(entity);
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.post('/', async (req, res) => {
  const entity = req.body;
  try {
    res.status(200).json(await EmployeeTerritoryService.create(entity));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.put('/:id', async (req, res) => {
  const entity = req.body;
  const { id } = req.params;
  try {
    res.status(200).json(await EmployeeTerritoryService.update(Number(id), entity));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.delete('/:id', async (req, res) => {
  const { id } = req.params;
  try {
    res.status(200).json(await EmployeeTerritoryService.delete(Number(id)));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

module.exports = router;

File: Go/GORM_ORM/api/employeeTerritoryAPI.go
package api

import (
	"net/http"
	"strconv"

	"GORM_ORM/bll/services"
	"GORM_ORM/dal/entities"

	"github.com/gin-gonic/gin"
)

type EmployeeTerritoryAPI struct {
	Service services.EmployeeTerritoryService
}

func NewEmployeeTerritoryAPI(s services.EmployeeTerritoryService) *EmployeeTerritoryAPI {
	return &EmployeeTerritoryAPI{Service: s}
}

func (h *EmployeeTerritoryAPI) GetAll(c *gin.Context) {
	entities, err := h.Service.GetAll()
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, entities)
}

func (h *EmployeeTerritoryAPI) GetByID(c *gin.Context) {
	id, _ := strconv.Atoi(c.Param("id"))
	entity, err := h.Service.GetByID(id)
	if err != nil {
		c.JSON(http.StatusNotFound, gin.H{"error": "EmployeeTerritory not found"})
		return
	}
	c.JSON(http.StatusOK, entity)
}

func (h *EmployeeTerritoryAPI) Create(c *gin.Context) {
	var entity entities.EmployeeTerritory
	if err := c.ShouldBindJSON(&entity); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	if err := h.Service.Create(&entity); err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusCreated, entity)
}

func (h *EmployeeTerritoryAPI) Update(c *gin.Context) {
	id, _ := strconv.Atoi(c.Param("id"))
	var entity entities.EmployeeTerritory
	if err := c.ShouldBindJSON(&entity); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	entity.EmployeeID = id
	if err := h.Service.Update(&entity); err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, entity)
}

func (h *EmployeeTerritoryAPI) Delete(c *gin.Context) {
	id, _ := strconv.Atoi(c.Param("id"))
	if err := h.Service.Delete(id); err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.Status(http.StatusOK)
}
